src/linear/coordinate_common.h

```diff
@@ -126,12 +122,11 @@ inline std::pair<double, double> GetGradientParallel(int group_idx, int num_grou
 inline std::pair<double, double> GetBiasGradientParallel(int group_idx, int num_group,
                                                          const std::vector<GradientPair> &gpair,
                                                          DMatrix *p_fmat) {
-  const RowSet &rowset = p_fmat->BufferedRowset();
```

Comment:
Can you provide your reasoning behind removing `BufferedRowset`? Is it the case that DMatrix always stores every row in `buffered_rowset_`?

Reply:
Yes every row is always used after we removed prob_buf_row.